### DISCUSSION TITLE
Inconsistent spacing between viz gallery charts

### DISCUSSION THREAD

**Discussion by michael-s-molina** [apache/superset#32962] [Ideas]
## Description

Depending on the chart's name length, we have inconsistent spacing between viz gallery charts. 

## Screenshot

<img width="1785" alt="Screenshot 2024-05-31 at 10 36 49" src="https://github.com/apache/superset/assets/70410625/b0064e56-be55-4693-bc9b-dc150382a4b9">
<img width="1784" alt="Screenshot 2024-05-31 at 10 37 08" src="https://github.com/apache/superset/assets/70410625/5b78081e-5721-4cb3-a91c-f79d4f1e414a">

## Design input
@kasiazjc I'm looking for a proposal where we consider the different chart names length when calculating the spacing between charts.

**Comment by rusackas** [comment]
May relate to https://github.com/apache/superset/issues/28792

**Comment by kasiazjc** [comment]
> ## Description
> Depending on the chart's name length, we have inconsistent spacing between viz gallery charts.
> 
> ## Screenshot
> <img alt="Screenshot 2024-05-31 at 10 36 49" width="1785" src="https://private-user-images.githubusercontent.com/70410625/335603498-b0064e56-be55-4693-bc9b-dc150382a4b9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIyNTM1MDIsIm5iZiI6MTcyMjI1MzIwMiwicGF0aCI6Ii83MDQxMDYyNS8zMzU2MDM0OTgtYjAwNjRlNTYtYmU1NS00NjkzLWJjOWItZGMxNTAzODJhNGI5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI5VDExNDAwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI2NGQxMmFkYTdjNjc2NmIzZGJhMmIxMjFiNWE0MThlZmQ3OWVhZDJjMGFjYmRkOGQyY2MzYmY2NGZhYjc5ZGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.23o7T3KXycOyWL1xK9U7TsyZDKsWoZFMqlfD31iNwG4"> <img alt="Screenshot 2024-05-31 at 10 37 08" width="1784" src="https://private-user-images.githubusercontent.com/70410625/335603516-5b78081e-5721-4cb3-a91c-f79d4f1e414a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIyNTM1MDIsIm5iZiI6MTcyMjI1MzIwMiwicGF0aCI6Ii83MDQxMDYyNS8zMzU2MDM1MTYtNWI3ODA4MWUtNTcyMS00Y2IzLWE5MWMtZjc5ZDRmMWU0MTRhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI5VDExNDAwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcyN2I1ZmNiZTllZGY0M2EyODQ0MjkxZjk0NWVkNDVmNjYyYjc0YzZmMjI1ZjI2OTFiZGVjMWVhZTVlNTU5NWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.YAveJvvru1nvZvfkEyGaJNIeGr8NwQiZ46q7R3Rw-8I">
> ## Design input
> @kasiazjc I'm looking for a proposal where we consider the different chart names length when calculating the spacing between charts.

Late to the party, sorry! 

But I am thinking - if chart name is longer than 2 lines we should truncate with "..." and show full name on hover. Two lines could also be a spacing that we use to keep everything consistent if that makes sense. Thoughts?

**Comment by rusackas** [comment]
Just moved this to an Ideas discussion, since it's not really a bug. But honestly, I think the inconsistent spacing bugs me less than truncation would. 

Another option that exists would be to make them all equally spaced, but that would create unnecessary gutter space for most rows, which is equally icky in my mind. 

Or, each row could be truncated when >2 lines, but "slide open" to reveal full names on hover... but that leads to layout shift, which is even grosser. 

Or we could NOT show names, and only show them in another means (tooltip/popover) on hover - more room for thumbnails I guess - but that would be confusing, and cause accessibility issues. 

I'm kind of thinking the current implementation may be the best of a bad situation.
